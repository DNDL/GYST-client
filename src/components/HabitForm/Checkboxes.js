import React from 'react';

export default function Checkboxes() {
  return (
    <fieldset>
      <legend>Track on which days:</legend>
      <label>
        S
        <input type="checkbox" name="sunday" value="sunday" />
      </label>
      <label>
        M
        <input type="checkbox" name="monday" value="monday" />
      </label>
      <label>
        T
        <input type="checkbox" name="tueday" value="tueday" />
      </label>
      <label>
        W
        <input type="checkbox" name="wednesday" value="wednesday" />
      </label>
      <label>
        Th
        <input type="checkbox" name="thursday" value="thursday" />
      </label>
      <label>
        F
        <input type="checkbox" name="friday" value="friday" />
      </label>
      <label>
        S
        <input type="checkbox" name="saturday" value="saturday" />
      </label>
    </fieldset>
  );
}
