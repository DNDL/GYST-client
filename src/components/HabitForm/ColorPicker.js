import React from 'react';

export default function ColorPicker() {
  return (
    <fieldset>
      <legend>Color: </legend>
      <label>
        Red
        <input type="radio" name="color" value="red" />
      </label>
      <label>
        Yellow
        <input type="radio" name="color" value="yellow" />
      </label>
      <label>
        Blue
        <input type="radio" name="color" value="blue" />
      </label>
    </fieldset>
  );
}
