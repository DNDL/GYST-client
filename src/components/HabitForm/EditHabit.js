import React, { Component } from 'react';
import { connect } from 'react-redux';

class EditHabit extends Component {

}