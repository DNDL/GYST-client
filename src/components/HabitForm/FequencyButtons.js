import React from 'react';

export default function ColorPicker() {
  return (
    <fieldset>
      <legend>Frequency:</legend>
      <label>
        Daily
        <input type="radio" name="frequency" value="daily" />
      </label>
      <label>
        Weekly
        <input type="radio" name="frequency" value="weekly" />
      </label>
      <label>
        Monthly
        <input type="radio" name="frequency" value="monthly" />
      </label>
    </fieldset>
  );
}
