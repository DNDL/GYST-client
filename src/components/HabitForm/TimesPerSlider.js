import React, { useState } from 'react';

export default function TimesPerSlider() {
  const [value, updateValue] = useState(10);

  return (
    <fieldset>
      <legend>Times per day</legend>
      <input
        type="range"
        min="1"
        max="10"
        onChange={({ target }) => updateValue(target.value)}
      ></input>
      <p>{value}</p>
    </fieldset>
  );
}

// TODO: slider default to 1, currently at 10
