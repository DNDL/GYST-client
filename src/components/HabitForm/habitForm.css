@import url('https://fonts.googleapis.com/css?family=Averia+Serif+Libre%7CJulius+Sans+One&display=swap');

.HabitForm {
  overflow: auto;
  display: flex;
  flex-direction: column;
  font-family: 'Julius Sans One', sans-serif;
  max-width: 360px;
  margin: 0 auto;

  label {
    margin: 3px;
    font-family: 'Julius Sans One', sans-serif;
  }

  span {
    font-family: 'Julius Sans One', sans-serif;
  }

  fieldset {
    border: 1px solid black;
    border-radius: 10px;
    padding-bottom: 0px;
    margin-bottom: 3px;
  }
  
  .checkbox {
    text-align: center;
    display: flex;
    justify-content: space-evenly; 
    margin: 0;
  }

  .radio {
    text-align: center;
    display: flex;
    justify-content: space-evenly; 
    
  }

  .update {
    background-color: rgb(167,219,216);
  }

  .create {
    background-color: rgb(240,120,24);
  }

}