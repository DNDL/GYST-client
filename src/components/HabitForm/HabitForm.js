import React from 'react';
import PropTypes from 'prop-types';
import Checkboxes from './Checkboxes';
import Slider from './Slider';

function HabitForm({ habitName }) {
  return (
    <form>
      <input
        type="text"
        name={habitName}
        value={habitName}
        placeholder="Habit name..."
      ></input>

      <fieldset>
        <legend>Frequency:</legend>
        <label>
          Daily
          <input type="radio" name="frequency" value="daily" />
        </label>
        <label>
          Weekly
          <input type="radio" name="frequency" value="weekly" />
        </label>
        <label>
          Monthly
          <input type="radio" name="frequency" value="monthly" />
        </label>
      </fieldset>

      <Slider />

      <Checkboxes />
    </form>
  );
}

HabitForm.propTypes = {
  habitName: PropTypes.string.isRequired
};

export default HabitForm;
