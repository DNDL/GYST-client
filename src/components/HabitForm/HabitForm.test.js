import React from 'react';
import { shallow } from 'enzyme';
import HabitForm from './HabitForm';

describe('HabitForm component', () => {
  it('renders HabitForm', () => {
    const wrapper = shallow(<HabitForm handleSubmit={() => {}} history={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
