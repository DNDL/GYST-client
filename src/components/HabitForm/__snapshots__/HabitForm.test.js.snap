// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitForm component renders HabitForm 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
